blueprints/automation/tado_temp_offset.yaml

```diff
@@ -43,35 +45,30 @@ blueprint:
         boolean:
 
     night_switch:
-      name: 'Nighttime (optional)'
-      description: 'At nighttime the sync rate is reduced to every 2 hours'
-      default:
+      name: "Nighttime (optional)"
+      description: "At nighttime the sync rate is reduced to every 2 hours"
       selector:
         entity:
-          domain: 
+          domain:
             - binary_sensor
             - input_boolean
-          
+
   source_url: https://gist.github.com/clemensao/homeassistant/blob/main/blueprints/automation/tado_temp_offset.yaml
+
 variables:
-  target_tado: !input 'target_tado'
-  source_temp_sensor: !input 'source_temp_sensor'
-  heating_threshold: !input 'heating_threshold'
-  night_switch: !input 'night_switch'
-  trigger_sensor: !input 'trigger_sensor'
-  tado_temp: '{{ state_attr(target_tado, "current_temperature") | float(19) }}'
+  target_tado: !input "target_tado"
+  source_temp_sensor: !input "source_temp_sensor"
+  heating_threshold: !input "heating_threshold"
+  night_switch: !input "night_switch"
+  trigger_sensor: !input "trigger_sensor"
+  tado_temp: '{{ state_attr(target_tado, "current_temperature") | float }}'  # Fetch dynamically
   current_offset: '{{ state_attr(target_tado, "offset_celsius") }}'
-  actual_temp: '{{ states(source_temp_sensor) | float(19) }}'
-  run_flag: '{{ night_switch == none or (night_switch != none and is_state(night_switch, "off")) }}'
-  heating_rate: >
-    {% set entities = device_entities(device_id("climate.schlafzimmer") ) %}
-    {% for s in entities %}
-      {% if 'heating' in s %}
-        {{ states(s) }}
-      {% endif %}
-    {% endfor %}
+  actual_temp: "{{ states(source_temp_sensor) | float }}"
+  run_flag: '{{ night_switch is not none and is_state(night_switch, "off") }}'
+  heating_rate: '{{ state_attr(target_tado, "heating_rate") | float }}'  # Adjust based on Tado's attributes
```

Comment:
This won't work in my configuration. heating_rate is no attribute of the climate entity but a seperate sensor-entity connected to the device. By default, the tado integration creates this sensor with an entity_id ending on "heating".

I suggest this approach:
`heating_rate: '{{ expand(device_entities(device_id(target_tado))) | selectattr("entity_id","match", "^sensor.*heating$") | map(attribute="state") | first | float }}'`

Reply:
I'm really new to this so it's great to learn! Agree as it threw up errors for me.